### PR TITLE
Adds Material Girders

### PR DESCRIPTION
**Pull request by Mechoid**
Adds material girders.
-Girder decon time is now partially dependant on material. Stronger girders take slightly longer. Steel girders are, actually, one tick faster to decon than before due to this.
-Wall girders are now under the integrity-based construction listing. Material-Name Wall Girder.
-Explosive-resistant girders in walls have a chance to survive as an unanchored girder, if their wall is destroyed.
-Radioactive girders affect the completed wall's radioactivity.
-Stronger materials make Stronger girders, with Steel at 201 health, and Durasteel at around 1000.

### CODE REVIEW

**Comment by elgeonmb** [comment]
Does this mean I can finally gun my way through a wall

**Comment by Atermonera** [comment]
Can you make wood girders, are they airtight, and do they burn?

**Comment by Anewbe** [comment]
If the girders are tougher than the walls of the same material, that needs to be adjust down hard.

**Comment by Mechoid** [comment]
You already could gun your way through a wall, physical rounds just to less damage.

**Comment by Mechoid** [comment]
Well, I don't know why steel walls are so wimpy and that their girders are already stronger than the wall itself.

**Comment by Mechoid** [comment]
Oh my lord I hate Git.

**Comment by Mechoid** [comment]
Wood girders and the like do not burn aside from their temperature resistance threshold as per normal material objects.

File: code/game/objects/structures/girders.dm [hunk 1]
```diff
@@ -130,14 +214,16 @@
 
 /obj/structure/girder/proc/take_damage(var/damage)
 	health -= damage
-
 	if(health <= 0)
 		dismantle()
+	else
```
Comment (Atermonera) [inline code/game/objects/structures/girders.dm]: +=

File: code/game/objects/structures/girders.dm [hunk 2]
```diff
@@ -174,36 +260,36 @@
...
+	to_chat(user, "<span class='notice'>You added reinforcement!</span>")
 
 	reinf_material = M
 	reinforce_girder()
 	return 1
 
 /obj/structure/girder/proc/reinforce_girder()
 	cover = reinf_material.hardness
-	health = 500
+	health = health + round(reinf_material.integrity/2)
```
Comment (Atermonera) [inline code/game/objects/structures/girders.dm]: +=

**Pull request merged**